sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -347,6 +347,14 @@ abstract class OutputWriterFactory extends Serializable {
       bucketId: Option[Int], // TODO: This doesn't belong here...
       dataSchema: StructType,
       context: TaskAttemptContext): OutputWriter
+
+  /**
+   * Returns a new instance of [[OutputWriter]] that will write data to the given path.
+   * @since 2.0.0
+   */
```

Comment:
Can we clarify the difference between these (i.e. which one is old and which is new?).